source/structuring.rst

```diff
@@ -14,402 +14,468 @@ functions is better than copying-and-pasting duplicate bits of code
-fetching may fetch ``person.json`` from
-``http://foo.bar/schemas/person.json``, even if ``address.json`` was loaded from
-somewhere else, such as the local filesystem. The drafts do not define this
-area of behaviour very clearly, and validator implementations may vary in
-exactly how they try to locate the referenced schema.
+``$ref`` can be used anywhere a schema is expected. When an object
+contains a ``$ref`` property, the object is considered a reference,
+not a schema. Therefore, any other properties you put there will not
+be treated as JSON Schema keywords and be ignored by the validator.
 
+For this example, let's say we want to define a customer record, where
+each customer may have both a shipping and a billing address.
+Addresses are always the same---they have a street address, city and
+state---so we don't want to duplicate that part of the schema
+everywhere we want to store an address.  Not only would that make the
+schema more verbose, but it makes updating it in the future more
+difficult.  If our imaginary company were to start doing international
+business in the future and we wanted to add a country field to all the
+addresses, it would be better to do this in a single place rather than
+everywhere that addresses are used.
 
-|draft6|
+.. schema_example::
 
-.. draft_specific::
+    {
+      "$id": "https://example.com/schemas/customer",
 
-    --Draft 4
-    In Draft 4, ``$id`` is just ``id`` (without the dollar sign).
+      "type": "object",
+      "properties": {
+        "first_name": { "type": "string" },
+        "last_name": { "type": "string" },
+        "shipping_address": { "$ref": "/schemas/address" },
+        "billing_address": { "$ref": "/schemas/address" }
+      },
+      "required": ["first_name", "last_name", "shipping_address", "billing_address"]
+    }
 
-The ``$id`` property should never be the empty string or an empty fragment
-(``#``), since that doesn't really make sense.
+The relative URIs in ``$ref`` resolves against the schema's identifier
+(``https://example.com/schemas/customer``) which results in
+``https://example.com/schemas/address``. The implementation retrieves
+that schema and evaluates the "shipping_address" and "billing_address"
+properties using that schema.
 
-Using $id with $ref
-```````````````````
+.. index::
+   single: definitions
+   single: structuring; definitions
+
+.. _definitions:
 
-``$id`` also provides a way to refer to subschema without using JSON Pointer.
-This means you can refer to them by a unique name, rather than by where they
-appear in the JSON tree.
+definitions
+-----------
 
-Reusing the address example above, we can add an ``$id`` property to the
-address schema, and refer to it by that instead.
+Sometimes we have small subschemas that are only intended for use in
+the current schema and it doesn't make sense to define them as
+separate schemas. Although we can identify any subschema using JSON
+Pointers or anchors, the ``definitions`` keyword gives us a
```

Comment:
Was using  `definitions` rather than `$defs` intentional, here?

Reply:
I meant `definitions`. So far, this site only covers draft-07. Introducing bits of draft 2019-09/2020-12 would be confusing at this point. Once all of these cleanup/fixes are completed, my next step is to update for draft 2019-09, then draft 2020-12. At that point I'll rename `definitions` to `$defs`.